common/tools/dev-tool/src/commands/run/bundle.ts

```diff
@@ -0,0 +1,147 @@
+import cjs from "@rollup/plugin-commonjs";
+import sourcemaps from "rollup-plugin-sourcemaps";
+import nodePolyfills from "rollup-plugin-polyfill-node";
+import json from "@rollup/plugin-json";
+import multiEntry from "@rollup/plugin-multi-entry";
+
+import { leafCommand, makeCommandInfo } from "../../framework/command";
+import { resolveProject, resolveRoot } from "../../util/resolveProject";
+import { createPrinter } from "../../util/printer";
+import { makeOnWarnForTesting, sourcemapsExtra } from "../../config/rollup.base.config";
+
+const log = createPrinter("bundle");
+
+export const commandInfo = makeCommandInfo(
+  "bundle",
+  "bundle a package using the default settings",
+  {
+    production: {
+      kind: "boolean",
+      default: true,
+      description: "build a CommonJS production bundle",
+    },
+    "browser-test": {
+      kind: "boolean",
+      default: true,
+      description: "build a bundle for browser testing",
+    },
+    "polyfill-node": {
+      kind: "boolean",
+      default: true,
+      description: "include a polyfill for Node.js builtin modules",
+    },
+  }
+);
+
+export default leafCommand(commandInfo, async (options) => {
+  const info = await resolveProject(process.cwd());
+
+  if (!info.packageJson.module) {
+    log.error(info.name, "does not specify a `module` field.");
+    return false;
+  }
+
+  const basePath = path
+    .relative(process.cwd(), path.dirname(path.parse(info.packageJson.module).dir))
+    .split(path.sep)
+    .join("/");
+
+  if (options.production) {
+    const baseConfig: rollup.RollupOptions = {
+      // Use the package's module field if it has one
+      input: info.packageJson.module,
+      external: [
+        ...nodeBuiltins,
+        ...Object.keys(info.packageJson.dependencies),
+        ...Object.keys(info.packageJson.devDependencies),
+      ],
+      preserveSymlinks: false,
+      plugins: [nodeResolve(), sourcemaps()],
+    };
+
+    try {
+      const bundle = await rollup.rollup(baseConfig);
+
+      await bundle.write({
+        file: "dist/index.js",
+        format: "cjs",
+        sourcemap: true,
+        exports: "named",
+      });
+    } catch (error) {
+      log.error(error);
+      return false;
+    }
+
+    log.success("Created production CommonJS bundle.");
+  }
+
+  if (options["browser-test"]) {
+    const pnpmStore = path
+      .relative(
+        process.cwd(),
+        path.join(await resolveRoot(), "common", "temp", "node_modules", ".pnpm")
+      )
+      .split(path.sep)
+      .join("/");
+
+    log.debug("Computed PNPM store relative path:", pnpmStore);
+
+    // Get a glob for a package name in the PNPM store
+    const globFromStore = (name: string): string =>
+      [pnpmStore, name.split("/").join("+"), "@*", "**/*.js"].join("/");
+
+    const browserTestConfig = {
+      input: {
+        include: [[basePath, "test", "**", "*.spec.js"].join("/")],
+        exclude: [[basePath, "test", "**", "node", "**"].join("/")],
```

Comment:
Question:

For browser-only tests, is "browser" folder path excluded in node test config?

Reply:
Yes, but rollup doesn't get itself involved in that. We run those tests from TS sources or from the ESM artifacts, so it's up to the mocha command that's used to exclude the browser files.

Reply:
makes sense.